specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/openshiftclusters/preview/2024-08-12-preview/redhatopenshift.json

```diff
@@ -0,0 +1,2732 @@
+          "type": "string"
+        }
+      }
+    },
+    "OpenShiftClusterProperties": {
+      "description": "OpenShiftClusterProperties represents an OpenShift cluster's properties.",
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The cluster provisioning state."
+        },
+        "clusterProfile": {
+          "$ref": "#/definitions/ClusterProfile",
+          "description": "The cluster profile."
+        },
+        "consoleProfile": {
+          "$ref": "#/definitions/ConsoleProfile",
+          "description": "The console profile."
+        },
+        "servicePrincipalProfile": {
+          "$ref": "#/definitions/ServicePrincipalProfile",
+          "description": "The cluster service principal profile."
+        },
+        "platformWorkloadIdentityProfile": {
+          "$ref": "#/definitions/PlatformWorkloadIdentityProfile",
+          "description": "The workload identity profile."
+        },
+        "networkProfile": {
+          "$ref": "#/definitions/NetworkProfile",
+          "description": "The cluster network profile."
+        },
+        "masterProfile": {
+          "$ref": "#/definitions/MasterProfile",
+          "description": "The cluster master profile."
+        },
+        "workerProfiles": {
+          "description": "The cluster worker profiles.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WorkerProfile"
+          },
+          "x-ms-identifiers": []
+        },
+        "workerProfilesStatus": {
+          "description": "The cluster worker profiles status.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/WorkerProfile"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "apiserverProfile": {
+          "$ref": "#/definitions/APIServerProfile",
+          "description": "The cluster API server profile."
+        },
+        "ingressProfiles": {
+          "description": "The cluster ingress profiles.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IngressProfile"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "OpenShiftClusterUpdate": {
+      "description": "OpenShiftCluster represents an Azure Red Hat OpenShift cluster.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "$ref": "#/definitions/Tags",
+          "description": "The resource tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/OpenShiftClusterProperties",
+          "description": "The cluster properties.",
+          "x-ms-client-flatten": true
+        },
+        "identity": {
+          "$ref": "../../../../../../common-types/resource-management/v6/managedidentity.json#/definitions/ManagedServiceIdentity",
```

Comment:
LintDiff `PatchBodyParametersSchema`: Step 2

Contains a required property `type` on L70:

https://github.com/Azure/azure-rest-api-specs/blob/8af8cd76b7349f357b8ea4f948ce2ee2aac318bb/specification/common-types/resource-management/v6/managedidentity.json#L41-L72

Reply:
Thanks @mikeharder. Should we not be referencing the common-types file for OpenShiftClusterUpdate?

I ask because it seems like if we pull it out, we will need to redefine ManagedServiceIdentity in our swagger file, which will result in duplicate schema errors in SDK generation.

Reply:
I don't know.  You can try searching the repo for specs similar to yours, to see what they do.  Or, you can ask at the ARM Office Hours: https://aka.ms/armofficehoursinfo

Reply:
Thanks - after looking at how other RPs do this, I've suppressed the failure with the same justification provided by them. Our swagger spec for `Update` is the same as other RPs, and, we get the patch body from common-types and the type is indeed patchable, so this appears to be a false positive.